lib/net.js

```diff
@@ -267,7 +276,7 @@ function Socket(options) {
   this._writableState.decodeStrings = false;
 
   // default to *not* allowing half open sockets
-  this.allowHalfOpen = options && options.allowHalfOpen || false;
+  this.allowHalfOpen = options.allowHalfOpen || false;
```

Comment:
… and, as I understand it, would allow us to get rid of this, if we just transform `allowHalfOpen` correctly to begin with?

Also, I don’t know what the motivation for this was, but I don’t get why the default is not allowing half-open connections.

Reply:
This is unrelated but `options` is always an object so there is no reason to see if it's truthy.